test.js

```diff
@@ -83,3 +83,17 @@ test('works with circular dependencies', t => {
 	t.is(childrenCalls, 4);
 	clearModule.all();
 });
+
+test('clear with filter', t => {
+	const testModuleFunctions = t;
+	const id = './fixture-filter';
+	require(id);
+
+	const parentmodule = Object.keys(require.cache).find(value => value.includes('parent-module'));
+
+	testModuleFunctions.not(typeof require.cache[parentmodule] !== 'undefined', false);
+	clearModule(id, {
+		filter: absolutePathToCachedFile => !absolutePathToCachedFile.match(/parent-module/)
```

Comment:
`.match` is not the correct method for checking.

Reply:
it a test for the example.  To ensure it works.